wso2-jenkins-jacoco-plugin/src/main/java/hudson/plugins/jacoco/ResourceFilesCollector.java

```diff
@@ -0,0 +1,139 @@
+    private static final String SOURCE_DIRECTORY = "sources";
+    private static final String JACOCO_RESOURCES = "jacocoResources";
+    private static final String RESOURCE_ZIP_FILE_NAME = "resources.zip";
+    private static final String JACOCO_EXEC_FILE = "jacoco.exec";
+    private static final int BYTE_BUFFER_LENGTH = 1024;
+
+    /**
+     * This method collect jacoco source,class files and
+     */
+    public void collectResourceFiles() {
+
+        final PrintStream logger = taskListener.getLogger();
+
+        if (isCreateZipFiles) {
+            logger.println("[JaCoCo plugin] Collecting .java , .class and jacoco.exec files");
+            final List<File> jacocoExecFiles = action.getJacocoReport().getExecFiles();
+            int fileSize = jacocoExecFiles.size();
+            if (fileSize == 0) {
+                logger.println("[JaCoCo plugin] No jacoco.exec file recorded");
+            } else {
+                String jobInfoDirectory = run.getRootDir().getAbsolutePath();
+                logger.println("[JaCoCo plugin] Creating folder to store files");
+                String jacocoResourcesFolder = jobInfoDirectory + File.separator + JACOCO_RESOURCES;
+                Path path = Paths.get(jacocoResourcesFolder);
+                try {
+                    Files.createDirectories(path);
+                    logger.println("[JaCoCo plugin] File created successfully at " + jobInfoDirectory);
+                    FileUtils.copyDirectory(new File(jobInfoDirectory + File.separator + "jacoco" + File.
+                            separator + CLASS_DIRECTORY), new File
+                            (jacocoResourcesFolder + File.separator + "classes"));
+                    FileUtils.copyDirectory(new File(jobInfoDirectory + File.separator + "jacoco" + File.
+                            separator + SOURCE_DIRECTORY), new File
+                            (jacocoResourcesFolder + File.separator + "sources"));
+                    if (fileSize == 1) {
+                        FileUtils.copyFile(jacocoExecFiles.get(0), new File(jacocoResourcesFolder +
+                                File.separator + JACOCO_EXEC_FILE));
+                    } else {
+                        logger.println("[JaCoCo plugin] Found " + fileSize + " jacoco.exec files.Merging...");
+                        ExecFileLoader loader = new ExecFileLoader();
+                        for (File exec : jacocoExecFiles) {
+                            loader.load(exec);
+                        }
+                        logger.println("[JaCoCo plugin] Fuck Successfully merged jacoco.exec files");
+                        loader.save(new File(jacocoResourcesFolder + File.separator + JACOCO_EXEC_FILE),
+                                false);
+                    }
+                    logger.println("[JaCoCo plugin] Compressing Jacoco resource folder...");
+                    compressAndCopy(jacocoResourcesFolder, jobInfoDirectory + File.separator +
```

Comment:
What's the reason for doing this?

Reply:
To calculate the line coverage we need to collect all the jacoco exec files,sources and class files and create a zip file.This creates a zip file inside the build folder containing all the above mentioned files.